flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestServerEndpointITCase.java

```diff
@@ -530,6 +530,20 @@ public void testDefaultVersionRouting() throws Exception {
 		}
 	}
 
+	@Test
+	public void testNonSslRedirectForEnabledSsl() throws Exception {
+		Assume.assumeTrue(config.getBoolean(SecurityOptions.SSL_REST_ENABLED));
+		OkHttpClient client = new OkHttpClient.Builder().followRedirects(false).build();
+		String httpsUrl = serverEndpoint.getRestBaseUrl() + "/path";
+		String httpUrl = httpsUrl.replace("https://", "http://");
+		Request request = new Request.Builder().url(httpUrl).build();
+		try (final Response response = client.newCall(request).execute()) {
+			assertEquals(HttpResponseStatus.MOVED_PERMANENTLY.code(), response.code());
+			assertTrue(response.headers().names().contains("Location"));
```

Comment:
error message could be improved by using hamcrest `assertThat(response.headers().names(), contains("Location"))`